amaranth_soc/csr/field.py

```diff
@@ -0,0 +1,188 @@
+from amaranth import *
+
+from .reg import Field
+
+
+__all__ = ["R", "W", "RW", "RW1C", "RW1S"]
+
+
+class R(Field):
+    __doc__ = Field._doc_template.format(
+    description="""
+    A read-only field.
+    """.strip(),
+    parameters="",
+    attributes="""
+    r_data : Signal(shape)
+        Read data. Drives the :attr:`~FieldPort.r_data` signal of ``port``.
+    """.strip())
+
+    def __init__(self, shape):
```

Comment:
This should take `reset` like `field.RW`.

One utility is to easily hard-code system parameters. A register which reads out the gateware version could have its reset value set to the version. Then no additional logic is needed to set the field data and it wouldn't happen to change if the system resets.

Reply:
This would work *only* in the case where no logic is driving the field. In that case, an explicit `.eq` is more clear as to what the intent is, exactly as efficient, and takes only one line of code.

Reply:
Point taken, but I don't like how the `.eq` is far away from the register definition in that case.

Another utility would be an active low status flag. Otherwise you'd have to introduce another signal in your logic to get the "defaults to 1" behavior.

Reply:
I agree it would be useful for active low status flags.

Reply:
For active-low flags that are also read-only, I think that either connecting a signal with `reset=1` to `action.R` or making a custom `FieldAction` are acceptable.
I don't think such flags are common enough for us to provide a built-in, but am happy to add one later if they are.